lib/rubocop/cop/factory_bot/redundant_factory_option.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module FactoryBot
+      # Remove redundant `factory` option.
+      #
+      # @example
+      #   # bad
+      #   association :user, factory: :user
+      #
+      #   # good
+      #   association :user
+      class RedundantFactoryOption < ::RuboCop::Cop::Base
+        extend AutoCorrector
+
+        include RangeHelp
+
+        MSG = 'Remove redundant `factory` option.'
+
+        RESTRICT_ON_SEND = %i[association].freeze
+
```

Comment:
Looking at those node patterns it feels like it would miss if more than just the assoxiation name is passed as a positional argument. FB seems to allow traits:

```ruby
factory :post do
  association :author, :admin, factory: :user, name: 'John Doe'
  # or
  association :author, factory: [:user, :admin], name: 'John Doe'
end
```

and it still can be offensive as in:
```ruby
  association :user, :admin, factory: :user
```